src/ja/user/cli.py

```diff
@@ -1,19 +1,210 @@
+        username: str = None if "username" not in data_loaded else data_loaded["username"]
+        password: str = None if "password" not in data_loaded else data_loaded["password"]
+        key_path: str = None if "key_path" not in data_loaded else data_loaded["key_path"]
+        passphrase: str = None if "passphrase" not in data_loaded else data_loaded["passphrase"]
+        email: str = None if "email" not in data_loaded else data_loaded["email"]
+
+        parser: ArgumentParser = ArgumentParser("JobAdder: add, cancel or a query a job.")
+        # shared options
+        parser.add_argument("--verbosity", "-v", type=int, choices=[0, 1, 2], default=verbosity)
+        parser.add_argument("--hostname", "--host", type=str,
+                            required=hostname is None, default=hostname,
+                            help="The name of the host to connect to.")
+        parser.add_argument("--username", "--user", type=str,
+                            required=username is None, default=username,
+                            help="The name of the user to use for login.")
+        parser.add_argument("--password", "--pass", type=str, default=password, help="The password to use for login.")
+        parser.add_argument("--key-path", "--kp", type=str, default=key_path, help="The key pair to use for login.")
+        parser.add_argument("--passphrase", "--pp", type=str, default=passphrase,
+                            help="The passphrase to use for decrypting private keys.")
+
+        subparsers = parser.add_subparsers(dest="command")
+        # parser for add command.
+        parser_add = subparsers.add_parser("add")
+        parser_add.add_argument("--label", "-l", help="Label for the job.")
+        parser_add.add_argument("--source", "-s", required=True, help="Path of the source file containing the job.")
+        parser_add.add_argument("--priority", "-p",
+                                choices="low medium high urgent 0 1 2 3".split(),
+                                default="1")
+        parser_add.add_argument("--threads", "-t", type=int, default=-1,
+                                help="Assign a number of cpu threads required for your job.")
+        parser_add.add_argument("--memory", "-m",
+                                type=int, default=1,
+                                help="Assign a specific amount of memory to your job")
+        parser_add.add_argument("--preemptible", "--pe", action="store_true")
+        parser_add.add_argument("--blocking", "--bl", action="store_true")
+        parser_add.add_argument("--mount", nargs="+", type=str, default=[])
+        parser_add.add_argument("--email", "-e", required=email is None, default=email)
+        parser_add.add_argument("--special-resources", "--sr", nargs="+", default=[])
+        parser_add.add_argument("--owner", default=os.getuid(), type=int, help="Not to be used outside of testing!")
```

Comment:
I am using this command to add jobs with different owners in the tests, so I can test querying by owners

Reply:
I see.
The owner ID should be determined by the username specified in the SSHConfig object after the job has been transferred to the server.
When you are finished with this PR, please remove this argument and just give the Job some value like -1 for the owner ID.

Reply:
Couldn't I keep it? One test won't run without this option. It won't appear in the CLI since its help is suppressed, and if the user somehow finds a way to run it, it'll be overridden on the server.

Reply:
It's not the cleanest solution but considering you're already behind on schedule I think it's fine for you to do it like that.